cupy/cuda/memory.pyx

```diff
@@ -541,6 +541,19 @@ cpdef MemoryPointer malloc_managed(size_t size):
 cdef object _current_allocator = _malloc
 cdef object _thread_local = threading.local()
 
+
+def _get_thread_local_allocator():
+    try:
+        allocator = _thread_local.allocator
+    except AttributeError:
+        allocator = _thread_local.allocator = None
```

Comment:
Should this be modifying `_thread_local.allocator`?

Reply:
Yes, semantically the default value of `_thread_local.allocator` should be `None`.
Assiging this at the first access saves the `AttributeError` in the second and later time.

Reply:
Yeah it feels a little weird, but I understand why it's there. Keeping it that way makes sense.